public/r2/v1/swagger/swagger.json

```diff
@@ -184,6 +184,79 @@
                 }
             }
         },
+        "/namespaces/{namespaceID}/ruleset/bulk": {
+            "put": {
+                "tags":[
+                    "namespaces"
+                ],
+                "summary": "Bulk update endpoint for namespace changes",
+                "operationId": "bulkUpdateRuleSet",
+                "consumes": [
+                    "application/json"
+                ],
+                "produces": [
+                    "application/json"
+                ],
+                "parameters": [
+                    {
+                        "in": "path",
+                        "name": "namespaceID",
+                        "description": "The name of the namespace",
+                        "type": "string",
+                        "required": true
+                    },
+                    {
+                        "in": "body",
+                        "name": "ruleset-change",
+                        "description": "The updates to the ruleset",
+                        "required": true,
+                        "schema": {
+                        "$ref": "#/definitions/RuleSetChange"
+                        }
+                    },
+                    {
+                        "in": "body",
+                        "name": "rule-set-version",
+                        "description": "The ruleset version to update, for check and set operations",
+                        "required": false,
+                        "type": "integer"
+                    },
+                    {
+                        "in": "body",
+                        "name": "atomic",
+                        "description": "flag to apply the changes atomically",
+                        "required": true,
+                        "type": "boolean"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "The ruleset updates have been applied",
+                        "schema": {
+                            "$ref": "#/definitions/ApiResponse"
+                        }
+                    },
+                    "404": {
+                        "description": "No such namespace",
+                        "schema": {
+                            "$ref": "#/definitions/ApiResponse"
+                        }
+                    },
+                    "409": {
```

Comment:
When would this occur? This is a PUT. This implies that we are creating something via a POST

Reply:
A 409 is the case when you provide a rule-set-version that is stale.

Reply:
Should the description clarify this?

Reply:
I see. The description here is a little confusing. But I think a 409 is still relevant.